core/utils.go

```diff
@@ -0,0 +1,29 @@
+package lucy
+
+import (
+	"reflect"
+)
+
+func Marshal(v interface{}) map[string]interface{} {
+	vtype := reflect.TypeOf(v)
+	vvalue := reflect.ValueOf(v)
+
+	if vtype.Kind() != reflect.Struct {
+		vtype = reflect.TypeOf(v).Elem()
+		vvalue = reflect.ValueOf(v).Elem()
+	}
+
+	tagMap := make(map[string]interface{})
+
```

Comment:
`Explanation`: What happens if v is not a struct

Reply:
If v is not a struct, it maybe a map, slice or a pointer to a struct. In that case, the value at the base address of v is assumed to be the struct. Then it is marshalled further. In any case if there is any unwanted behaviour in marshalling, the tagMap will remain empty or the program will go into panic. This is an internal function and will not be used by the user directly.

Reply:
Necessary refactoring will be done.